### ISSUE TITLE
Replace `payum/payum` dependency with `payum/core`

### ISSUE THREAD

**Issue by rimas-kudelis** [Sylius/Sylius#14766]
Sylius currently depends on `payum/payum`, which is a monorepo package and provides several integrations out of the box (Authorize.net, Be2Bill, Klarna, Offline, Payex, Paypal, Sofort, Stripe, and even a gateway skeleton for developers to start from).

I think it would be better to depend on `payum/core` instead, which is a subtree split, containing only the core classes and none of the actual integrations. While `paym/payum` doesn't have additional dependencies (in fact, it has one less), it's still extra code which will often be left unused by Sylius installations. Depending on core only and asking actual payment method plugins to depend on specific integrations would be cleaner and just the right thing to do, in my opinion.

**Comment by coldic3** [comment]
Hey @rimas-kudelis,
your proposal seems reasonable to me. The PR with the replacement would be very welcome. However, we probably should mention this change in the upgrade file.

**Comment by jakubtobiasz** [comment]
Hello @rimas-kudelis 👋🏼!
I gave it a try to remove `payum/payum` and replace it by concrete packages, but:

1) It fails CI, I made some fixes, but e2e are still failing. It means somewhere is a code highly depending on a `payum/payum` availability, as it delivers all the needed deps.
2) Some packages depend on the fact Sylius delivers `payum/payum` package, and don't declare any of the Payum's packages as a dependency. It means additional effort not only for us, but also for plugin creators.
3) I don't see too much value added when you compare it to the required effort.

As no will of creating a PR appeared, I'm closing this issue. Thanks for the suggestion. It's not a bad idea, but in the current state it won't bring too much value :).

**Comment by rimas-kudelis** [comment]
Wouldn't it make more sense to set a target milestone to 2.0 instead of just closing this issue as "completed"?

As for the will to create a PR, I just didn't want to make promises that I wasn't sure I would find time to fulfil.

**Comment by jakubtobiasz** [comment]
I'll schedule it for a next maintenance session, but I hardly believe it will be fixes in the `2.0`. I'll let you know once we decide whether we want to keep this task or not.

**Comment by rimas-kudelis** [comment]
Your PR was failing on PayPal integration tests, so I added `payum/paypal-express-checkout-nvp` to the list of requirements and – eureka! – the tests no longer seem to fail _(EDIT: some seemingly unrelated tests timed out, not sure what you do in such cases)_.

It's funny 'cause none of the classes in this package are ever referenced from Sylius, but I guess the (now deprecated) PayPal Express Checkout integration relies on this functionality implicitly.

**Comment by Prometee** [comment]
@jakubtobiasz removing `payum/payum` requirements  is possible, you will have to replace it by :

```
composer require payum/core payum/stripe payum/paypal-express-checkout-nvp
```

It will breaks only one notify action for the Be2Bill which is not covered by SyliusPayumBundle because of this `if` statement :
https://github.com/Sylius/Sylius/blob/1.13/src/Sylius/Bundle/PayumBundle/Extension/UpdatePaymentStateExtension.php#L53
BTW, that's why I made a PR to make this extension works for any kind of Payum `Notify` requests : https://github.com/Sylius/Sylius/pull/12622
It forces me to add this extension for each Sylius Payum plugins I created or participated in.
To avoid creating any issue, adding the Payum Be2Bill package (`payum/be2bill`)  to the requirement is the way to fix all kind of issues.

PS: What can be considered for Sylius v2 is to remove supports for the legacy `payum/stripe` and `payum/paypal-express-checkout-nvp`.
